src/build_options.rs

```diff
@@ -469,6 +514,39 @@ fn split_extra_args(given_args: &[String]) -> Result<Vec<String>> {
     Ok(splitted_args)
 }
 
+fn is_cross_compiling(target: &Target) -> Result<bool> {
```

Comment:
This function diverges a lot from the pyo3 version because it's not used in build script and have no access to some build time environment variables.